mavros/src/plugins/global_position.cpp

```diff
@@ -325,7 +325,7 @@ class GlobalPositionPlugin : public plugin::PluginBase {
 						map_point.x(), map_point.y(), map_point.z());
 
 			// Set the current fix as the "map" origin if it's not set
-			if (!is_map_init) {
+			if (!is_map_init && fix->status.status == 0) {
```

Comment:
Are you sure? 0 means no fix, and from your description you need 3D fix at least.

Reply:
So I've thought about this a little bit more, and there seems to be 2 different properties to denote gps fix. there is gps fix status which I believe denotes if the drone has any GPS signal. And then there is gps fix_type which denotes which kind of fix it is. If we use fix_type, the value would need to > 2. You are right my commit above works in sim testing, but is not robust. if there is a condition where the initial GPS coordinates are erroneous this would not work properly, but it would still be better than the current version, with no check. That being said, making the check rely on fix_type is the better route and I will rework it. It will probably be at least a week since I just went on vacation and I don't have the time or the proper setup to test a new version of the fix. 
Thanks!

Reply:
No, code is fine, i just forgot that NavSatFix aready evaluate fix_type > 2 (meaning 3D FIX or augmented).
But it does not check is that augmented fix or not. So eq to 0 is good, but better to change it to:
```
&& fix->status.status >= sensor_msgs::NavSatStatus::STATUS_FIX
```

See: https://github.com/mavlink/mavros/blob/master/mavros/src/plugins/global_position.cpp#L163